waku/v2/peermanager/peer_manager.go

```diff
@@ -0,0 +1,121 @@
+		zap.Uint("maxRelayPeersValue", maxRelayPeersValue), zap.Uint("outRelayPeersTargetValue", outRelayPeersTargetValue),
+		zap.Uint("inRelayPeersTarget", pm.InRelayPeersTarget))
+
+	return pm
+}
+
+func (pm *PeerManager) SetHost(host host.Host) {
+	pm.host = host
+}
+
+// Start starts the processing to be done by peer manager.
+func (pm *PeerManager) Start() {
+	go pm.connectivityLoop()
+}
+
+// This is a connectivity loop, which currently checks and prunes inbound connections.
+func (pm *PeerManager) connectivityLoop() {
+
+	for {
+		pm.pruneInRelayConns()
+		time.Sleep(peerConnectivityLoopSecs * time.Second)
+	}
+}
+
+func (pm *PeerManager) filterPeersByProto(peers peer.IDSlice, proto protocol.ID) peer.IDSlice {
+	var filteredPeers peer.IDSlice
+	//TODO: This can be optimized once we have waku's own peerStore
+
+	for _, p := range peers {
+		supportedProtocols, err := pm.host.Peerstore().GetProtocols(p)
+		if err != nil {
+			pm.logger.Warn("Failed to get supported protocols for peer", zap.String("peerID", p.String()))
+		}
+		if slices.Contains(supportedProtocols, proto) {
+			filteredPeers = append(filteredPeers, p)
+		}
+	}
+	return filteredPeers
+}
+
+func (pm *PeerManager) pruneInRelayConns() {
+
+	var inRelayPeers peer.IDSlice
+	//Group peers by their connected direction inbound or outbound.
+	inPeers, outPeers, err := pm.host.Peerstore().(wps.WakuPeerstore).GroupPeersByDirection()
+	if err != nil {
+		return
+	}
+	pm.logger.Info("Number of peers connected", zap.Int("inPeers", inPeers.Len()), zap.Int("outPeers", outPeers.Len()))
+
+	//Need to filter peers to check if they support relay
+	inRelayPeers = pm.filterPeersByProto(inPeers, WakuRelayIDv200)
+	outRelayPeers := pm.filterPeersByProto(outPeers, WakuRelayIDv200)
+	pm.logger.Info("Number of Relay peers connected", zap.Int("inRelayPeers", inRelayPeers.Len()), zap.Int("outRelayPeers", outRelayPeers.Len()))
+
+	if inRelayPeers.Len() > int(pm.InRelayPeersTarget) {
+		//Start disconnecting peers, based on what?
+		//For now, just disconnect most recently connected peers
+		//TODO: Need to have more intelligent way of doing this, maybe peer scores.
+		pm.logger.Info("Number of in peer connections exceed targer relay peers, hence pruning", zap.Int("inRelayPeers", inRelayPeers.Len()), zap.Uint("inRelayPeersTarget", pm.InRelayPeersTarget))
+		for pruningStartIndex := pm.InRelayPeersTarget; pruningStartIndex < uint(inRelayPeers.Len()); pruningStartIndex++ {
+			p := inRelayPeers[pruningStartIndex]
+			err := pm.host.Network().ClosePeer(p)
+			if err != nil {
+				pm.logger.Warn("Failed to disconnect connection towards peer", zap.String("peerID", p.String()))
+			}
+			pm.host.Peerstore().RemovePeer(p) //TODO: Should we remove the peer immediately?
+			err = pm.host.Peerstore().(wps.WakuPeerstore).SetDirection(p, network.DirUnknown)
+			if err != nil {
+				pm.logger.Warn("Failed to remove metadata for peer", zap.String("peerID", p.String()))
+			}
```

Comment:
Looking at the implementation of `RemovePeer` in go-libp2p, it remove the metadata for the peers. Then i think it is not necessary necessary to set the direction to unknown in the next line as the record would have been removed from the DB anyways

Reply:
Thanks for catching that. Will change it.